### PR TITLE
TLS support for server side and relp protocol changes

### PR DESCRIPTION
**Pull request by dhlavac**
Added TLS support and some protocol changes

### CODE REVIEW

**Comment by dhlavac** [comment]
@jvymazal @radosroka  Please, look at that and try to test that

File: lib/relp/server.rb [hunk 1]
```diff
@@ -1,19 +1,25 @@
 require 'relp/relp_protocol'
 require 'logger'
 require 'thread'
+require "openssl"
 
 module Relp
   class RelpServer < RelpProtocol
-    def initialize(host, port, logger = nil, callback)
+    def initialize(port, callback, host = '0.0.0.0' , tls_context = nil, logger = nil)
```
Comment (jvymazal) [inline lib/relp/server.rb:8]: Why changing previous parameter order?

**Review by jvymazal — CHANGES_REQUESTED** [review]
Could you add something to changelog as well?
If you are making a major release this should be at least mentioned somewhere, better with justification why it was needed and what the backwards-(in)compatible changes are.

**Comment by dhlavac** [comment]
@jvymazal Changes done

**Review by jvymazal — APPROVED** [review]

**Pull request merged**